components/camel-quartz/src/test/java/org/apache/camel/routepolicy/quartz/CronScheduledRoutePolicyTest.java

```diff
@@ -330,10 +331,16 @@ public void configure() {
 
             ServiceHelper.suspendService(context.getRoute("test").getConsumer());
 
-            Thread.sleep(5000);
-            template.sendBody("direct:start", "Ready or not, Here, I come");
+            try {
+                Awaitility.await().timeout(5, TimeUnit.SECONDS).catchUncaughtExceptions().untilAsserted(() -> {
+                    template.sendBody("direct:start", "Ready or not, Here, I come");
+                    success.assertIsSatisfied();
+                });
+            } catch (CamelExecutionException e) {
+                throw new CamelExecutionException(e.getMessage(), e.getExchange(), e);
+            }
```

Comment:
Hm, this seems a bit strange to me. I think it would probably be better to separate error handling from the test assertion.